sigilrecorder.lic

```diff
@@ -0,0 +1,205 @@
+custom_require.call(%w[common common-arcana common-items common-travel drinfomon events spellmonitor equipmanager])
+
+class SigilRecorder
+  include DRC
+  include DRCA
+  include DRCI
+  include DRCT
+
+	def initialize
+		require 'yaml'
+		@data = YAML.load(File.read("./scripts/data/base-sigils.yaml"))
+		@city_list = ['shard', 'crossing', 'riverhaven', 'other']
+		@sigil_list = ['abolition','congruence','induction','permutation','rarefaction', 'antipode','ascension','clarification','decay','evolution','integration','metamorphosis','nurture','paradox','unity']
+		@settings = get_settings
+		get_region
+		get_season #make sure we're recording sigils to the correct season
+		region_room_list #generate the walk_list array
+		generate_nogo_list
+		
+		Flags.add('danger-room', /.*begins to advance on you.*|You weave back and forth in a dance-like motion|You must be closer to use tactical abilities on your opponent/)
+		@walk_list.each do |roomid|
+			@data['nogo'].include?(roomid) ? next : (travel_result = walk_to(roomid, false))
+			respond("Travel Debug: #{travel_result} | Room Target: #{roomid}")
+			if Room.current.id == nil
+			  respond("Adding #{roomid} to nogo due to lack of visible room number in LOOK.")
+			  add_to_nogo(roomid)
+			  unstuck_me(roomid)
+			  next
+			end
+			if travel_result == true
+				@current_room = Room.current.id
+				DRCA.do_buffs(@settings, 'outdoors')
+				if check_room_data < 2
+				  @room_sigils = []
+				  get_season
+				  if find_sigil?
+				    next #skip this room because something went wrong or it is dangerous
+				  end
+				  respond("This room contains the following sigils #{@room_sigils}")
+				else
+				  respond("Sigils in this room already known")
+				end #if
+			else
+				respond('Unable to reach target room. Adding to no-go list and moving to next room.')
+				add_to_nogo(Room.current.id)
+				next
+			end #if
+			
+			write_to_file
+		end #do
+	end #initialize
+		
+		
+	def get_season
+		/^It is currently (\w+) and it is.*$/ =~ bput('time', /^It is currently.*/)
+		@season = $1
+	end #get_season
+	
+	
+	def get_region
+		case Room.current.image.downcase
+		when /ilithi|shard/
+		  @region = 'shard'
```

Comment:
Rather than settings instance vars (`@`), I'd much prefer the function to return a value that can then be passed along later. It helps keep track of where vars are coming from, instead of having a bunch of instance vars that anything can modify.

What do you think?

Reply:
Makes sense, I'll see if I can untangle this and have get_region return the region name instead.